### PR TITLE
Added service_labele to match resource for attach labels for GCP

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**

### CODE REVIEW

File: cartography/intel/gcp/apigateway.py [hunk 1]
```diff
@@ -698,26 +698,24 @@ def sync(
...
+    label.sync_labels(neo4j_session, locations, gcp_update_tag,
+                      common_job_parameters, 'apigateway_locations', 'GCPLocation')
     # API Gateway APIs
     apis = get_apis(apigateway, project_id, regions)
     load_apis(neo4j_session, apis, project_id, gcp_update_tag)
     for api in apis:
         load_apis_entity_relation(neo4j_session, api, gcp_update_tag)
     # Cleanup APIs
     cleanup_apis(neo4j_session, common_job_parameters)
-    label.sync_labels(neo4j_session, apis, gcp_update_tag, common_job_parameters, 'apis')
```
Comment (mpurusottamc) [inline cartography/intel/gcp/apigateway.py:709]: we should call cleanup for apis as well.

File: cartography/intel/gcp/apigateway.py [hunk 2]
```diff
@@ -698,26 +698,24 @@ def sync(
...
     # API Gateway API Configs
     configs = get_api_configs(apigateway, project_id, regions)
     load_api_configs(neo4j_session, configs, project_id, gcp_update_tag)
     # Cleanup API Gateway Configs
     cleanup_api_configs(neo4j_session, common_job_parameters)
-    label.sync_labels(neo4j_session, configs, gcp_update_tag, common_job_parameters, 'api configs')
```
Comment (mpurusottamc) [inline cartography/intel/gcp/apigateway.py:715]: we should call cleanup for api configs as well.

File: cartography/intel/gcp/apigateway.py [hunk 3]
```diff
@@ -698,26 +698,24 @@ def sync(
...
     # API Gateway Gateways
     gateways = get_gateways(apigateway, project_id, regions)
     load_gateways(neo4j_session, gateways, project_id, gcp_update_tag)
     for gateway in gateways:
         load_gateway_entity_relation(neo4j_session, gateway, gcp_update_tag)
     # Cleanup API Gateway Gateways
     cleanup_api_gateways(neo4j_session, common_job_parameters)
-    label.sync_labels(neo4j_session, gateways, gcp_update_tag, common_job_parameters, 'apigateways')
```
Comment (mpurusottamc) [inline cartography/intel/gcp/apigateway.py:723]: we should call cleanup for api gateways as well.

File: cartography/intel/gcp/dns.py [hunk 4]
```diff
@@ -249,10 +249,9 @@ def sync(
...
         dns_zones = zones_list
     load_dns_zones(neo4j_session, dns_zones, project_id, gcp_update_tag)
-    label.sync_labels(neo4j_session, dns_zones, gcp_update_tag, common_job_parameters, 'dns_zones')
+    label.sync_labels(neo4j_session, dns_zones, gcp_update_tag, common_job_parameters, 'dns_zones', 'GCPDNSZone')
     # RECORD SETS
     dns_rrs = get_dns_rrs(dns, dns_zones, project_id)
     load_rrs(neo4j_session, dns_rrs, project_id, gcp_update_tag)
     # TODO scope the cleanup to the current project - https://github.com/lyft/cartography/issues/381
     cleanup_dns_records(neo4j_session, common_job_parameters)
-    label.sync_labels(neo4j_session, dns_rrs, gcp_update_tag, common_job_parameters, 'dns record')
```
Comment (mpurusottamc) [inline cartography/intel/gcp/dns.py:258]: we should call cleanup for dns record as well.

File: cartography/intel/gcp/label.py [hunk 5]
```diff
@@ -54,20 +53,28 @@ def _load_labels_tx(tx: neo4j.Transaction, data: List[Dict], update_tag: int) ->
...
         labels_list = get_labels_list(data)
         if len(labels_list) > 0:
-            logger.info(f"Loading Labels for {service}")
-            load_labels(neo4j_session, labels_list, update_tag)
+            logger.info(f"Loading Labels for {service_name}")
+            common_job_parameters['service_label'] = service_label
+            load_labels(neo4j_session, labels_list, update_tag, common_job_parameters)
+            cleanup_labels(neo4j_session, common_job_parameters, service_name)
+            try:
+                del common_job_parameters["service_label"]
```
Comment (mpurusottamc) [inline cartography/intel/gcp/label.py:78]: Should we delete `service_name` as well?

File: cartography/graph/job.py [hunk 6]
```diff
@@ -74,12 +74,15 @@ def as_dict(self) -> Dict:
...
 
     @classmethod
-    def from_json(cls, blob: str, short_name: str = None):
+    def from_json(cls, blob: str, short_name: str = None, parameters: Dict = None,):
         """
         Create a job from a JSON blob.
         """
         data: Dict = json.loads(blob)
         statements = _get_statements_from_json(data, short_name)
+        if parameters.get('service_label', None):
```
Comment (mpurusottamc) [inline cartography/graph/job.py:83]: can we avoid changing graph/job.py? is it possible to implement this outside this file.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
minor changes necessary.

File: cartography/intel/gcp/apigateway.py (hunk 1 above)
Comment (ganeshjagdale9545) [reply cartography/intel/gcp/apigateway.py:709]: cleanup are called cleanup methods not deleted

File: cartography/intel/gcp/label.py (hunk 5 above)
Comment (ganeshjagdale9545) [reply cartography/intel/gcp/label.py:78]: yes because every cleanup job has separate `service_name`

File: cartography/graph/job.py (hunk 6 above)
Comment (ganeshjagdale9545) [reply cartography/graph/job.py:83]: no because if we pass label name directly in cleanup job query then it takes as string and gives error

Comment (ganeshjagdale9545) [reply cartography/graph/job.py:83]: there is no anther way to use this logic

**Pull request merged**